pkg/operator/controller/nodeobservability/daemonset_test.go

```diff
@@ -162,19 +591,319 @@ func TestEnsureDaemonset(t *testing.T) {
 				},
 			}
 
-			gotExist, _, err := r.ensureDaemonSet(context.TODO(), nodeObs, sa, test.OperatorNamespace)
+			_, err := r.ensureDaemonSet(context.TODO(), nodeObs, sa, r.Namespace)
+			if err != nil {
+				t.Fatalf("unexpected error received: %v", err)
+			}
+
+			ds := &appsv1.DaemonSet{}
+			err = r.Client.Get(context.Background(), types.NamespacedName{Namespace: test.TestNamespace, Name: daemonSetName}, ds)
+			if err != nil {
+				t.Fatalf("failed to get daemonset: %v", err)
+			}
+			if diff := cmp.Diff(ds, tc.expectedDS); diff != "" {
+				t.Errorf("resource mismatch:\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestUpdateDaemonSet(t *testing.T) {
+
+	for _, tc := range []struct {
+		name              string
+		existingDaemonset *appsv1.DaemonSet
+		desiredDaemonset  *appsv1.DaemonSet
+		expectedDaemonset *appsv1.DaemonSet
+		expectUpdate      bool
+	}{
+		{
+			name: "image changed",
+			existingDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v1").
+					build(),
+				).build(),
+			desiredDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v2").
+					build(),
+				).build(),
+			expectedDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v2").
+					build(),
+				).build(),
+			expectUpdate: true,
+		},
+		{
+			name: "container args changed",
+			existingDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v1").
+					withArgs([]string{"--arg1=1"}...).
+					build(),
+				).build(),
+			desiredDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v2").
+					withArgs([]string{"--arg=value"}...).
+					build(),
+				).build(),
+			expectedDaemonset: testDaemonset("daemonset", "test-namespace", "test-sa").
+				withContainers(testContainer("agent", "agent:v2").
+					withArgs([]string{"--arg=value"}...).
+					build(),
+				).build(),
+			expectUpdate: true,
+		},
+		{
+			name: "container injected into daemonset",
```

Comment:
Can you please add a test case for injected volume/volumemounts too?

Reply:
They are already present, changed the test name to make it more clear.